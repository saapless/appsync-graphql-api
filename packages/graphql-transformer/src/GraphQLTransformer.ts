export class GraphQLTransformer {
  constructor() {}

  public transform() {
    return "transformed";
  }
}
