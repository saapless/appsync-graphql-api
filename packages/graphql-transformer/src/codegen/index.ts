export * from "./ast";
export * from "./printer";
export * from "./document";
