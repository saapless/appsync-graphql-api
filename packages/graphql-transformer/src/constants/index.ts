export const ConnectionDirective = Object.freeze({
  HAS_ONE: "hasOne",
  HAS_MANY: "hasMany",
});

export const ModelDirective = Object.freeze({
  MODEL: "model",
});

export const UtilityDirective = Object.freeze({
  SERVER_ONLY: "serverOnly",
  CLIENT_ONLY: "clientOnly",
  READ_ONLY: "readOnly",
  WRITE_ONLY: "writeOnly",
  FILTER_ONLY: "filterOnly",
  INTERNAL: "internal",
});

export const ScalarType = Object.freeze({
  STRING: "String",
  INT: "Int",
  FLOAT: "Float",
  BOOLEAN: "Boolean",
  ID: "ID",
  LONG: "Long",
});

export const RelationType = Object.freeze({
  ONE_TO_ONE: "oneToOne",
  ONE_TO_MANY: "oneToMany",
  MANY_TO_MANY: "manyToMany",
});
