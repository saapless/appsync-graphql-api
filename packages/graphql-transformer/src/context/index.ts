export * from "./TransformerContext";
export type {
  DescriptorOperation,
  Key,
  KeyOperator,
  KeyValue,
  OperationActionType,
  ResolverDescriptor,
  ReturnType,
} from "./ResolverLoader";
