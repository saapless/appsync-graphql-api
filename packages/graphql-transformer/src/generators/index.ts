export { ResolverGeneratorBase } from "./ResolverGeneratorBase";
export { DynamoDbGenerator } from "./DynamoDBGenerator";
export { NoneGenerator } from "./NoneGenerator";
export { SchemaTypesGenerator } from "./SchemaTypesGenerator";
export { ResolverTypesGenerator } from "./ResolverTypesGenerator";
