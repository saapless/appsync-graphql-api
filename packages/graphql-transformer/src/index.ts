export * from "./transformer";
export { definitionFromSource } from "./utils";
