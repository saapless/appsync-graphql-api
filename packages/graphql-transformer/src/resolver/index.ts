export { ResolverBase } from "./ResolverBase";
export { FieldResolver } from "./FieldResolver";
export { FunctionResolver } from "./FunctionResolver";
