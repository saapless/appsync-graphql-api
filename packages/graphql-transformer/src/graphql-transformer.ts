import { GraphQLTransformer } from "./GraphQLTransformer";

export function createTransformer() {
  return new GraphQLTransformer();
}

export function executeTranformer() {}
