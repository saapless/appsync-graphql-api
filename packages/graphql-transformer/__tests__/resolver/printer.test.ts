import { printAST, DocumentDefinition, NodeKind, _literal } from "../../src/resolver";

const ast = {
  _kind: NodeKind.CODE_DOCUMENT,
  body: [
    {
      _kind: NodeKind.IMPORT_DECLARATION,
      from: "@aws-appsync/utils",
      specifiers: [
        {
          _kind: NodeKind.MODULE_NAMED_SPECIFIER,
          value: {
            _kind: NodeKind.LITERAL,
            value: "Context",
            type: "string",
          },
        },
      ],
    },
  ],
  // requestFunction: {
  //   kind: NodeKind.FUNCTION_DEFINITION,
  //   body: { kind: NodeKind.CODE_BLOCK, value: "return {};" },
  //   exports: true,
  //   name: "request",
  //   parameters: [
  //     {
  //       kind: NodeKind.FUNCTION_PARAMETER,
  //       name: "ctx",
  //       type: "Context",
  //     },
  //   ],
  // },
  // responseFunction: {
  //   kind: NodeKind.FUNCTION_DEFINITION,
  //   body: { kind: NodeKind.CODE_BLOCK, value: "return ctx.result;" },
  //   exports: true,
  //   name: "response",
  //   parameters: [
  //     {
  //       kind: NodeKind.FUNCTION_PARAMETER,
  //       name: "ctx",
  //       type: "Context",
  //     },
  //   ],
  // },
} satisfies DocumentDefinition;

describe("printAST", () => {
  describe("on FunctionParameter", () => {
    it("print parameter without type", () => {
      const value = printAST(_literal("ctx"));

      expect(value).toBe("ctx");
    });

    // it("print parameter with type", () => {
    //   const value = printAST({
    //     kind: NodeKind.FUNCTION_PARAMETER,
    //     name: "ctx",
    //     type: "Context",
    //   });

    //   expect(value).toBe("ctx: Context");
    // });

    // it("print parameter with default", () => {
    //   const value = printAST({
    //     kind: NodeKind.FUNCTION_PARAMETER,
    //     name: "value",
    //     default: "true",
    //   });

    //   expect(value).toBe("value = true");
    // });
  });

  it("should print the AST", () => {
    const result = printAST(ast);
    expect(result).toMatchSnapshot();
  });
});
