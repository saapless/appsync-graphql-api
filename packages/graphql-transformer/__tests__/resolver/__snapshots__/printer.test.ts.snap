// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`printAST should print the AST 1`] = `""`;
